src/kixi/datastore/metrics.clj

```diff
@@ -107,37 +107,41 @@
     [influx-reporter registry]  
     component/Lifecycle
     (start [component]
-      (let [with-reg (update component :registry #(or %
-                                                      (let [reg (new-registry)]
-                                                        (jvm/instrument-jvm reg)
-                                                        reg)))
-            reg (:registry with-reg)]
-        (-> with-reg
-            (update :reporter #(or %
-                                   (let [reporter (influxdb/reporter reg 
-                                                                     influx-reporter)]
-                                     (log/info "Starting InfluxDb Metrics Reporter")
-                                     (influxdb/start reporter (:seconds influx-reporter))
-                                     reporter)))
-            (update :meter-mark #(or %
-                                     (meter-mark! reg)))
-            (update :insert-time-in-ctx #(or %
-                                             (insert-time :resource)))
-            (update :expose-metrics-resource #(or %
-                                                  (expose-metrics-resource reg)))
-            (update :record-ctx-metrics #(or %
-                                             (record-metrics reg :resource))))))
+      (if-not registry
+        (let [with-reg (update component :registry #(or %
```

Comment:
more and more wondering if this kind of stuff doesn't belong in a lib, since I basically lifted out your code to use metrics in heimdall, so if improvements are required I should propagate those as well (and same for all metrics-issuing apps).